spyder/widgets/variableexplorer/utils.py

```diff
@@ -292,17 +295,7 @@ def value_to_display(value, minmax=False):
         elif isinstance(value, NUMERIC_TYPES) or isinstance(value, bool) or \
           isinstance(value, datetime.date):
             display = repr(value)
-        elif isinstance(value, int64):
-            display = repr(value)
-        elif isinstance(value, int32):
-            display = repr(value)
-        elif isinstance(value, float64):
-            display = repr(value)
-        elif isinstance(value, float32):
-            display = repr(value)
-        elif isinstance(value, complex_) or isinstance(value, complex128) :
-            display = repr(value)
-        elif isinstance(value, complex64):
+        elif True in [isinstance(value, x) for x in numeric_type_list]:

```

Comment:
I'd write this as `any(isinstance(value, x) for x in numeric_type_list)` or `isinstance(value, numeric_type_tuple)`; in the latter case, I think it needs to be a tuple of types, not a list of types.